exist-core/src/main/java/org/exist/security/internal/Password.java

```diff
@@ -152,25 +152,27 @@ public boolean equals(Object obj) {
     	if(obj == this) {
             return true;
         }
-    	
-    	if(obj == null) {
-            return false;
-        }
-    	
-    	if(obj instanceof Password p) {
 
-            if(algorithm != p.algorithm) {
-                throw new RuntimeException("Cannot compare passwords with different algorithms i.e. " + algorithm + " and " + p.algorithm);
+        switch (obj) {
```

Comment:
Could this be rewritten to a return switch statement?
```java
return switch (obj) { case null, default -> false … }

Reply:
yes indeed